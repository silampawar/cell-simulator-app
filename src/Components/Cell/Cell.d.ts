import { UpdateStatusFunction } from "../CellSimulator/CellSimulator.d";

export interface ICellProps {
    isAlive: boolean
    updateStatus:UpdateStatusFunction
    currentPos:{x:number,y:number}
}