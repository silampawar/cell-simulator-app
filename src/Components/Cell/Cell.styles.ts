import styled from "styled-components"

export const Container = styled.div`
    background-color: ${(props: { isAlive: boolean })=>props.isAlive?'green':'grey'};
    border: 1px solid;
    height: 60px;
    width: 60px;
`
export const Anchor = styled.a`
text-decoration: none;
 cursor: default; 
 color: blanchedalmond;
`