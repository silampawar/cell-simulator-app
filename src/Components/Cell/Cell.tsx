import {Anchor, Container} from "./Cell.styles"
import {ICellProps} from "./Cell.d"


const Cell = ({ isAlive,updateStatus,currentPos }:ICellProps): JSX.Element => {

    return <><Anchor data-testid= "cell-anchore-itemid" href="#" onClick={()=>updateStatus({currentStatus: isAlive,gridPos:currentPos})}>
                <Container data-testid= "cell-container-itemid" isAlive={isAlive} ></Container>
                </Anchor></>
}

export default Cell;