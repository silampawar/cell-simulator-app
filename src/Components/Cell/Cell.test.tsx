import Cell from './Cell';
import 'jest-styled-components'
import { render, screen } from '@testing-library/react';

test('renders Cell component link- alive cell', () => {
  render(<Cell isAlive={true} updateStatus={jest.fn()} currentPos={{x:2,y:3 }}/>)
  let element = screen.getAllByTestId('cell-container-itemid')
  expect(element.length).toBe(1);

  expect(screen.getByTestId('cell-container-itemid')).toHaveStyle(`background-color: green`)

});

test('renders Cell component link- dead cell', () => {
  render(<Cell isAlive={false} updateStatus={jest.fn()} currentPos={{x:2,y:3 }}/>)
  expect(screen.getByTestId('cell-container-itemid')).toHaveStyle(`background-color: grey`)

});

