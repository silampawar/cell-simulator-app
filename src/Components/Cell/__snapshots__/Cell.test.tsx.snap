// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cell component link 1`] = `ShallowWrapper {}`;
